sqlglot/dialects.py

```diff
@@ -317,12 +317,21 @@ class Hive(Dialect):
     DATE_FORMAT = "'yyyy-MM-dd'"
     TIME_FORMAT = "'yyyy-MM-dd HH:mm:ss'"
 
+    class HiveMap(exp.Map):
+        is_var_len_args = True
+
     def _map_sql(self, expression):
-        keys = expression.args["keys"].args["expressions"]
-        # values is an array because map is var_len_args due to hive dialect
-        values = expression.args["values"][0].args["expressions"]
+        keys = expression.args["keys"]
+        values = expression.args["values"]
+
+        if not isinstance(keys, exp.Array) or not isinstance(values, exp.Array):
+            self.unsupported(
+                "Cannot convert array columns into map use SparkSQL instead."
```

Comment:
I don't know the exact relationships between SparkSQL and HiveQL, but is SparkSQL always an available replacement for HiveQL?

Reply:
no, it's not, but hive is deprecated. sparksql is a superset of hivesql